src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,101 @@
+                return new DeleteCommand(taskIndex);
+            }
+            if (fullCommand.length() >= 6 && fullCommand.substring(0, 4).equals("todo")) {
+                String description = fullCommand.substring(5, fullCommand.length());
+                return new AddCommand(Task.TaskType.TODO, description, null, null);
+            }
+            if (fullCommand.length() >= 25 && fullCommand.substring(0, 8).equals("deadline")) {
+                String description = fullCommand.substring(9, fullCommand.indexOf("/") - 1);
+                String dateAndTime = fullCommand.substring(fullCommand.indexOf("/") + 4, fullCommand.length());
+                try {
+                    LocalDate date = LocalDate.parse(
+                            dateAndTime.substring(0, 10),
+                            DateTimeFormatter.ofPattern("dd-MM-yyyy"));
+                    LocalTime time = null;
+                    if (dateAndTime.length() > 10) {
+                        time = LocalTime.parse(
+                                dateAndTime.substring(dateAndTime.indexOf(" ") + 1, dateAndTime.length()),
+                                DateTimeFormatter.ofPattern("HHmm"));
+                    }
+                    return new AddCommand(Task.TaskType.DEADLINE, description, date, time);
+                } catch (DateTimeParseException e) {
+                    throw new DukeException("Cannot parse this command, " +
+                            "please enter correct date and time format: dd-MM-yyyy HHmm");
+                }
+            }
+            if (fullCommand.length() >= 22 && fullCommand.substring(0, 5).equals("event")) {
+                String description = fullCommand.substring(6, fullCommand.indexOf("/") - 1);
+                String dateAndTime = fullCommand.substring(fullCommand.indexOf("/") + 4, fullCommand.length());
+                try {
+                    LocalDate date = LocalDate.parse(
+                            dateAndTime.substring(0, 10),
+                            DateTimeFormatter.ofPattern("dd-MM-yyyy"));
+                    LocalTime time = null;
+                    if (dateAndTime.length() > 10) {
+                        time = LocalTime.parse(
+                                dateAndTime.substring(dateAndTime.indexOf(" ") + 1, dateAndTime.length()),
+                                DateTimeFormatter.ofPattern("HHmm"));
+                    }
+                    return new AddCommand(Task.TaskType.EVENT, description, date, time);
+                } catch (DateTimeParseException e) {
+                    throw new DukeException("Cannot parse this command, " +
+                            "please enter correct date and time format: dd-MM-yyyy HHmm");
+                }
+            }
+            if (fullCommand.length() >= 6 && fullCommand.substring(0, 4).equals("find")) {
+                String findKeyword = fullCommand.substring(fullCommand.indexOf(" "), fullCommand.length());
```

Comment:
Should findKeyword be named keywordToFind instead, so that it sounds like a noun instead of a verb?